hal/api/toolchain.h

```diff
@@ -207,6 +207,10 @@
  *  Mark a function declaration as deprecated, if it used then a warning will be
  *  issued by the compiler possibly including the provided message. Note that not
  *  all compilers are able to display the message.
+ *
+ *  @note
```

Comment:
An arbitrary date is fine and anyone who gets the warning wont have the context as to when it was first marked as depreciated, just why it is from the warning. If we wanted to enforce automatically we could assert when the date expires but I don't think its the right time for that. Just marked as could be removed at anytime after the as printed by the warning